pkg/apis/proxy/aws.go

```diff
@@ -0,0 +1,71 @@
+package proxy
+
+import (
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+	"github.com/gin-gonic/gin"
+
+	"github.com/seal-io/walrus/pkg/dao/model"
+	optypes "github.com/seal-io/walrus/pkg/operator/types"
+)
+
+type awsProxy struct {
+	conn    *model.Connector
+	context *gin.Context
+}
+
+func (p *awsProxy) handle(requestPath, rawQuery string) {
+	requestPath = appendSchemaIfMissing(requestPath)
+
+	u, err := url.Parse(requestPath)
+	if err != nil {
+		_ = p.context.AbortWithError(http.StatusBadRequest, err)
+		return
+	}
+
+	if !isWhiteListDomain(u.Hostname()) {
```

Comment:
Prefer to add whiteList as the attribute of the awsProxy or a function, the whiteList for aws proxy, should just include aws domain.

Reply:
updated